orchagent/pfcwdorch.cpp

```diff
@@ -320,15 +302,61 @@ void PfcWdOrch<DropHandler, ForwardHandler>::unregisterFromWdDb(const Port& port
         sai_object_id_t queueId = port.m_queue_ids[i];
 
         // Unregister in syncd
-        PfcWdOrch<DropHandler, ForwardHandler>::getPfcWdTable()->del(sai_serialize_object_id(queueId));
+        m_pfcWdTable->del(sai_serialize_object_id(queueId));
+        m_entryMap.erase(queueId);
     }
 }
 
 template <typename DropHandler, typename ForwardHandler>
-PfcWdSwOrch<DropHandler, ForwardHandler>::PfcWdSwOrch(DBConnector *db, vector<string> &tableNames):
-    PfcWdOrch<DropHandler, ForwardHandler>(db, tableNames)
+PfcWdSwOrch<DropHandler, ForwardHandler>::PfcWdSwOrch(
+        DBConnector *db,
+        vector<string> &tableNames,
+        vector<sai_port_stat_t> portStatIds,
+        vector<sai_queue_stat_t> queueStatIds):
+    PfcWdOrch<DropHandler,
+    ForwardHandler>(db, tableNames),
+    m_pfcWdDb(new DBConnector(PFC_WD_DB, DBConnector::DEFAULT_UNIXSOCKET, 0)),
+    m_pfcWdTable(new ProducerStateTable(m_pfcWdDb.get(), PFC_WD_STATE_TABLE)),
+    c_portStatIds(portStatIds),
+    c_queueStatIds(queueStatIds)
 {
     SWSS_LOG_ENTER();
+
+    string platform = getenv("platform") ? getenv("platform") : "";
```

Comment:
in /usr/bin/start.sh swss docker

export platform=`sonic-cfggen -v platform`

platform = $onie_platform, not sure if this work.

Reply:
https://github.com/Azure/sonic-buildimage/blob/master/dockers/docker-orchagent/orchagent.sh#L7

Reply:
In orchagent.sh it's exported as onie_switch_asic